Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -639,30 +642,26 @@ void WebProcessProxy::shutDown()
     WEBPROCESSPROXY_RELEASE_LOG(Process, "shutDown:");
 
     if (m_isInProcessCache) {
-        processPool().webProcessCache().removeProcess(*this, WebProcessCache::ShouldShutDownProcess::No);
+        protectedProcessPool()->checkedWebProcessCache()->removeProcess(*this, WebProcessCache::ShouldShutDownProcess::No);
         ASSERT(!m_isInProcessCache);
     }
 
     shutDownProcess();
 
-    if (m_webConnection) {
-        m_webConnection->invalidate();
-        m_webConnection = nullptr;
-    }
+    if (RefPtr webConnection = std::exchange(m_webConnection, nullptr))
+        webConnection->invalidate();
 
     m_backgroundResponsivenessTimer.invalidate();
     m_activityForHoldingLockedFiles = nullptr;
     m_audibleMediaActivity = std::nullopt;
     m_mediaStreamingActivity = std::nullopt;
     m_throttler.didDisconnectFromProcess();
 
-    for (auto& page : pages()) {
-        if (page)
-            page->disconnectFramesFromPage();
-    }
+    for (auto& page : pages())
```

Comment:
Use CheckedRef/Ref?

Reply:
pages() returns a `Vector<Ref<WebPageProxy>>`. I guess I can use `Ref&&` instead of `auto&`.

Reply:
`Ref&` or `Ref&&` doesn't build. I'll just use `Ref` then. It's a bit more ref counting churn than `auto&` though.